### PR TITLE
New: Overwrite File in schedule modal

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

Adds overwrite toggle to some formats (just gsheet for now)

## Proposed Changes

- Add overwrite toggle.

## Screenshots
Screen Shot 2021-11-22 at 4 05 03 PM

Screen Shot 2021-11-22 at 4 05 18 PM

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/common-actions/schedule.ts [hunk 1]
```diff
@@ -249,6 +256,24 @@ export default class ScheduleActionComponent extends Component<Args> {
...
   }
 
+  @action
+  toggleOverwriteFile() {
+    assert('The localDeliveryRule is defined', this.localDeliveryRule);
+    assert(
+      'Must be a type that supports overwriteFile in order to toggle',
+      this.overwriteableFormats.includes(this.localDeliveryRule.format.type)
+    );
+    const format = this.localDeliveryRule.format as GsheetFormat; //currently only type that supports this, make more dynamic with new types
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/common-actions/schedule.ts:266]: Looked into the typing around this a little bit and I think the reason visualizations work but this doesn't when using the type discriminator is that visualizations are typed via a mixin

https://github.com/yavin-dev/framework/blob/ee70aecb7e817229146d86a2b8b886d2c94051b5/packages/core/addon/mixins/models/has-visualization.ts#L15

**Review by kevinhinterlong — APPROVED** [review]

File: packages/reports/tests/integration/components/common-actions/schedule-test.js [hunk 2]
```diff
@@ -113,6 +113,8 @@ module('Integration | Component | common actions/schedule', function (hooks) {
     assert.dom('.schedule__modal-format-trigger').hasText('csv', '`.csv` is the default format value');
 
     assert.dom('.schedule__modal-rejected').doesNotExist('rejected error does not show');
+
+    assert.dom('.schedule__modal-overwrite').doesNotExist('Overwrite switch not shown for non overwriteable types');
```
Comment (daniellevinson) [inline packages/reports/tests/integration/components/common-actions/schedule-test.js:117]:
```suggestion
    assert.dom('.schedule__modal-overwrite').doesNotExist('Overwrite switch not shown for non overwritable types');
```

**Review by daniellevinson — APPROVED** [review]

**Comment by jkusa** [comment]
@cythrawll can we match the overwrite description font styles with the denali input description styles

**Comment by cythrawll** [comment]
Screen Shot 2022-01-26 at 1 59 12 PM

**Pull request merged**